### PR TITLE
Change default output type of CreatePeaksworkspace back to standard Peaksworkspace

### LINKED ISSUE #30984: Change CreatePeaksWorkspace default output type to Peaksworkspace
- Related Gitlab Task: [SCD265](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/265)

Based on the discussion in https://github.com/mantidproject/mantid/pull/30884#issuecomment-805067754, we need to switch the default output type of `CreatePeaksworkspace` back to standard Peaksworkspace such that existing workflows are not interrupted by the introduction of LeanElasticPeak and LeanElasticPeakWorkspace.

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**
- Related Gitlab Task: [265](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/265)
- This is #30983 into `master`

This PR adds a new option, `OutputType`, to `CreatePeaksworkspace`, which can be toggled between "Peak" and "LeanElasticPeak".
- "Peak": use standard Peaksworkspace as output, this is the DEFAULT option
- "LeanElasticPeak": use new lean elastic Peaksworkspace, which does not have an instrument attached to it

demo

> Note: the arguments in the screenshot is outed, please use "LeanElasticPeak" to indicate that the output type should be lean elastic peak workspace.

**To test:**
Testing script
```python
from mantid.simpleapi import *

#-- default, regular PeaksWorkspaces with an instrument specified
sampleWs = CreateSampleWorkspace()
pws_withInst = CreatePeaksWorkspace(
    InstrumentWorkspace=sampleWs,
    NumberOfPeaks=3,
    )

#-- defualt, regular Peaksworkspace without specifying an instrument
pws_withoutInst = CreatePeaksWorkspace(NumberOfPeaks=3)

#-- lean pws, Lean Elastic Peaksworkspace without specifying an instrument
lpws = CreatePeaksWorkspace(
    NumberOfPeaks=3,
    OutputType="LeanElasticPeak",
    )

#-- lean pws, Lean Elastic Peakworkspace with specifying an instrument, which is ignored
lpws_ignoreInst = CreatePeaksWorkspace(
    InstrumentWorkspace=sampleWs,
    NumberOfPeaks=3,
    OutputType="LeanElasticPeak",
    )
```

Fixes #30984. 

*This does not require release notes* because **we are switching back to the previous default behavior within the same release cycle.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/CreatePeaksWorkspace.cpp [hunk 1]
```diff
@@ -36,58 +37,76 @@ void CreatePeaksWorkspace::init() {
   declareProperty(std::make_unique<WorkspaceProperty<IPeaksWorkspace>>(
                       "OutputWorkspace", "", Direction::Output),
                   "An output workspace.");
+  // explicit control of output peak workspace tyep
+  // Full: standar peak workspace
+  // Lean: LeanElasticPeakWorkspace
+  const std::vector<std::string> peakworkspaceTypes{"Full", "Lean"};
+  declareProperty(
```
Comment (rosswhitfield) [inline Framework/Algorithms/src/CreatePeaksWorkspace.cpp:44]: We should be consistent with this property between different algorithms, for FindPeaksMD #30953 I had the property `OutputPeakType` with the options `Peak` and `LeanElasticPeak`. I will the same property with my upcoming changes for `PredictPeaks`.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]
There is a doc-test that is now failing. It might be nice to have the old doc-test before #30914 see https://github.com/mantidproject/mantid/pull/30914/files#diff-909c8931bcc4ef9c0c5ec8d2729bf048c65de6ae19b283ccaa56939b1ad56954R30

And a new one with LeanElasticPeaksWrokspace.

File: Framework/Algorithms/src/CreatePeaksWorkspace.cpp (hunk 1 above)
Comment (KedoKudo) [reply Framework/Algorithms/src/CreatePeaksWorkspace.cpp:44]: I will update the docs and option list as suggested.

**Review by rosswhitfield — APPROVED** [review]

**Comment by martyngigg** [comment]
Thanks for doing this 👍🏻

**Pull request merged**